### PR TITLE
Add some factories for PrimitiveInfo

### PR DESCRIPTION
**Pull request by mrobinson**
This will make it easier to add new members in the future (for hit
testing). We also make local_clip a LocalClip instead of an
Option<LocalClip> because this should never be None.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1708)

### CODE REVIEW

File: webrender/examples/basic.rs [hunk 1]
```diff
@@ -334,12 +313,7 @@ impl Example for App {
             let spread_radius = 0.0;
             let simple_border_radius = 8.0;
             let box_shadow_type = BoxShadowClipMode::Inset;
-            let info = LayoutPrimitiveInfo {
-                rect: rect,
-                local_clip: Some(LocalClip::from(bounds)),
-                is_backface_visible: true,
-            };
-
+            let info = LayoutPrimitiveInfo::new_with_clip_rect(rect, bounds);
```
Comment (kvark) [inline webrender/examples/basic.rs]: should probably just be `with_clip_rect`?

File: webrender/examples/blob.rs [hunk 2]
```diff
@@ -253,11 +249,7 @@ impl Example for App {
                                       api::MixBlendMode::Normal,
                                       Vec::new());
 
-        let info = api::LayoutPrimitiveInfo {
-            rect: (30, 30).by(500, 500),
-            local_clip: Some(api::LocalClip::from(bounds)),
-            is_backface_visible: true,
-        };
+        let info = api::LayoutPrimitiveInfo::new((30, 30).by(500, 500));
```
Comment (kvark) [inline webrender/examples/blob.rs]: missing local clip?

File: webrender/examples/blob.rs [hunk 3]
```diff
@@ -266,11 +258,7 @@ impl Example for App {
             blob_img1,
         );
 
-        let info = api::LayoutPrimitiveInfo {
-            rect: (600, 600).by(200, 200),
-            local_clip: Some(api::LocalClip::from(bounds)),
-            is_backface_visible: true,
-        };
+        let info = api::LayoutPrimitiveInfo::new((600, 600).by(200, 200));
```
Comment (kvark) [inline webrender/examples/blob.rs]: missing local clip?

File: webrender/examples/basic.rs (hunk 1 above)
Comment (mrobinson) [reply webrender/examples/basic.rs]: Okay. I'll make this change.

File: webrender/examples/blob.rs (hunk 2 above)
Comment (mrobinson) [reply webrender/examples/blob.rs]: I'll fix this.

File: webrender/examples/blob.rs (hunk 3 above)
Comment (mrobinson) [reply webrender/examples/blob.rs]: Ditto!

**Comment by mrobinson** [comment]
Okay. I've updated the branch with the changes and rebased it on top of the rustfmt changes.

**Comment by kvark** [comment]
Thanks @mrobinson , and sorry about the rebase pain!
@bors-servo r+

**Comment by staktrace** [comment]
@mrobinson This broke the gecko build, please see https://bugzilla.mozilla.org/show_bug.cgi?id=1400216#c5 and provide a gecko-side patch. Thanks!

**Comment by mrobinson** [comment]
@staktrace Thanks for the heads up. I've posted a patch to Bugzilla which should fix the build.

**Pull request merged**